R/gh.R

```diff
@@ -208,6 +211,24 @@ gh <- function(endpoint, ..., per_page = NULL, .token = NULL, .destfile = NULL,
   res
 }
 
+gh_response_length <- function(res) {
+  if (!is.null(names(res)) && length(res) > 1 &&
+      names(res)[1] == "total_count") {
+    # Ignore total_count, incomplete_results, repository_selection
+    # and take the first list element to get the length
+    lst <- vapply(res, is.list, logical(1))
+    nm <- setdiff(
+      names(res),
+      c("total_count", "incomplete_results", "repository_selection")
+    )
+    tgt <- which(lst[nm])[1]
+    if (is.na(tgt)) length(res) else length(res[[ nm[tgt] ]])
+    length(res[[2]])
+  } else {
```

Comment:
I think there needs to be a third case here for endpoints like `/user`

Reply:
The length is only really important if `.limit` is specified, and people will not do that for `/user`. 

We could do more heuristics, but I want to be cautious. So for `/user` we can essentially keep the old behavior, nothing changes imo.

Reply:
This seems reasonable, I will need to check that ghclass is not doing something silly like setting `.limit` for everything